src/WebSocketChannel/WebSocketChannel.cs

```diff
@@ -249,6 +249,13 @@ async ValueTask<ReadOnlyMemory<byte>> ReadCoreAsync(CancellationToken cancellati
                         if (received.Count == 0)
                             break;
 
+                        if (!received.EndOfMessage)
+                        {
+                            var tempMem = memoryOwner.Memory.Slice(0, count);
+                            memoryOwner = MemoryPool<byte>.Shared.Rent(count + 512);
+                            tempMem.CopyTo(memoryOwner.Memory);
```

Comment:
Should this cost only be incurred if the count is actually greater than the current mem size?

Reply:
Ah ok, you mean something like:
if (!received.EndOfMessage && received.Count == 512)
                        {
                            var tempMem = memoryOwner.Memory.Slice(0, count);
                            memoryOwner = MemoryPool<byte>.Shared.Rent(count + _bufferSize);
                            tempMem.CopyTo(memoryOwner.Memory);
                        }
   Correct?

Reply:
more like `count >= memoryOwner.Memory.Length` (or Count, don't recall). What matters is the global byte count, since the Memory.Slice(count) uses that to give the ReceiveAsync its max size.

Reply:
also, I just rebased on top of master, to get the build fixes from main

Reply:
PR Updated 
https://github.com/devlooped/WebSocketChannel/compare/main...corradocavalli:WebSocketChannel:main?expand=1